feat: multi-tenancy (#483)
eHealthAfrica/aether: README.md

- `GS_SECRET_ACCESS_KEY`: Google Cloud Secret Access Key.
  [How to create Access Keys on Google Cloud Storage](https://cloud.google.com/storage/docs/migrating#keys)

#### Multi-tenancy

- `MULTITENANCY`, Enables or disables the feature, is `false` if unset or set
  to empty string, anything else is considered `true`.

> - `DEFAULT_REALM`, `aether` The default realm for the objects created

Reviewer: ```suggestion
- `DEFAULT_REALM`, `aether` The default realm for artefacts created
```

>   while multi-tenancy was not enabled.

Reviewer: ```suggestion
  if multi-tenancy is not enabled.
```

- `REALM_COOKIE`, `aether-realm` The name of the cookie that keeps the current
  tenant id in the request headers.

These variables are included in the `.env` file. Change them to enable or disable
the multi-tenancy feature.

Example with multi-tenancy enabled:

```text
# .env file
MULTITENANCY=yes
DEFAULT_REALM=my-current-tenant
REALM_COOKIE=cookie-realm
```

```bash
./scripts/docker_start.sh
```

Example with multi-tenancy disabled:

```text
# .env file
MULTITENANCY=
```

```bash
./scripts/docker_start.sh
```

The technical implementation is explained in
[Multi-tenancy README](/aether-common-library/aether/common/multitenancy/README.md).

**Notes:**

- Everything is accessible to admin users in the `admin` section.

- In Aether Kernel:
  - All the schemas are accessible to all tenants.
  - The entities without project are not accessible using the REST API.

- In Aether ODK module:

>   - The surveyors could be assigned to more than one realm.

Reviewer: Is this still the case?

Author: Yes, but only in the admin section, using the REST API is hard to do because the user is not accessible.

- In Aether CouchDB-Sync module:
  - Devices and mobile users are accessible to all tenants.
  - The module is not fully ready for this feature. It's possible to run

>     an instance assigned to one realm but not to many realms without sharing

Reviewer: ```suggestion
    an instance assigned to a single realm but not to multiple realms without sharing
```
